### PR TITLE
ci: add new workflow for make test-short

### PR DESCRIPTION
**Pull request by tungbq**
## Overview
Closes: #2361 

Changes:
- Add a CI workflow for make test-short

**NOTE:**  The timeout in the issue was 60s but as the actual run take around 6mins (on my fork repo), so I add time out 10 mins for make test-short step at the beginning

Trial run successfully:
- See: https://github.com/tungbq/celestia-app/actions/runs/5988912031/job/16244574169?pr=5

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: .github/workflows/test-short.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: test-short
```
Comment (rootulp) [inline .github/workflows/test-short.yml:1]: [blocking] Proposal to add this job to the `test.yml` file that already exists at https://github.com/celestiaorg/celestia-app/blob/c90e61d5a2d0c0bd0e123df4ab416f6f0d141b7f/.github/workflows/test.yml

File: .github/workflows/test-short.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  test-short:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Run tests
```
Comment (rootulp) [inline .github/workflows/test-short.yml:18]: once we move this to test.yml, proposal to disambiguate the job names by something like

```suggestion
      - name: Run tests in short mode
```

**Review by rootulp — COMMENTED** [review]
Thanks for the contribution @tungbq !

**Comment by rootulp** [comment]
> The timeout in the issue was 60s but as the actual run take around 6mins (on my fork repo), so I add time out 10 mins for make test-short step at the beginning

On my machine `make test-short takes less than a minute`. Confirmed via:

```diff
## test-short: Run tests in short mode.
test-short:
	@echo "--> Running tests in short mode"
-	@go test -mod=readonly ./... -short
+	@time go test -mod=readonly ./... -short -count=1
.PHONY: test-short
```


Does the 6 mins refer to how long CI takes? If so, I think we can make the timeout in CI 10 minutes and add a 60 second timeout in the Makefile for the make test-short command via `-timeout 1m`.

File: .github/workflows/test-short.yml (hunk 1 above)
Comment (tungbq) [reply .github/workflows/test-short.yml]: Updated! Thanks

File: .github/workflows/test-short.yml (hunk 2 above)
Comment (tungbq) [reply .github/workflows/test-short.yml]: Updated! Thanks

**Comment by rootulp** [comment]
Yay make test-short passes on this PR: https://github.com/celestiaorg/celestia-app/actions/runs/6001720074/job/16276519961?pr=2367

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**